### PR TITLE
Key-Vault deployment

### PR DESCRIPTION
**Pull request by v-fearam**
Arm and bicep key-vault incorporation

### CODE REVIEW

File: azure_jumpstart_arcbox/ARM/azuredeploy.json [hunk 1]
```diff
@@ -88,14 +92,18 @@
     "capiTemplateUrl": "[uri(variables('templateBaseUrl'), 'ARM/kubernetes/ubuntuCapi.json')]",
     "clientVmTemplateUrl": "[uri(variables('templateBaseUrl'), 'ARM/clientVm/clientVm.json')]",
     "mgmtTemplateUrl": "[uri(variables('templateBaseUrl'), 'ARM/mgmt/mgmtArtifacts.json')]",
-    "mgmtStagingStorageUrl": "[uri(variables('templateBaseUrl'), 'ARM/mgmt/mgmtStagingStorage.json')]"
+    "mgmtStagingStorageUrl": "[uri(variables('templateBaseUrl'), 'ARM/mgmt/mgmtStagingStorage.json')]",
+    "keyVaultTemplateUrl": "[uri(variables('templateBaseUrl'), 'ARM/key-vault/key-vault.json')]"
```
Comment (dkirby-ms) [inline azure_jumpstart_arcbox/ARM/azuredeploy.json:96]: We are using camel case for other folder names so I suggest we keep that convention here and change the path from key-vault to keyVault.

File: azure_jumpstart_arcbox/ARM/key-vault/key-vault.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{
```
Comment (dkirby-ms) [inline azure_jumpstart_arcbox/ARM/key-vault/key-vault.json:1]: Change filename and path from key-vault to keyVault

File: azure_jumpstart_arcbox/bicep/key-vault/key-vault.bicep [hunk 3]
```diff
@@ -0,0 +1,40 @@
+@description('Azure service principal client id')
```
Comment (dkirby-ms) [inline azure_jumpstart_arcbox/bicep/key-vault/key-vault.bicep:1]: change filename and path from key-vault to keyVault

File: azure_jumpstart_arcbox/bicep/main.bicep [hunk 4]
```diff
@@ -109,3 +109,11 @@ module mgmtArtifactsAndPolicyDeployment 'mgmt/mgmtArtifacts.bicep' = {
     deployBastion: deployBastion
   }
 }
+
+module keyVaultDeployment 'key-vault/key-vault.bicep' = if (flavor == 'DevOps') {
```
Comment (dkirby-ms) [inline azure_jumpstart_arcbox/bicep/main.bicep:113]: change name from key-vault to keyVault

**Review by dkirby-ms — CHANGES_REQUESTED** [review]
Looks good, suggest changing keyvault path/name from key-vault to keyVault to align to other resources.

File: azure_jumpstart_arcbox/ARM/azuredeploy.json (hunk 1 above)
Comment (DCMattyG) [reply azure_jumpstart_arcbox/ARM/azuredeploy.json:96]: BTW, "keyVault" is Lower Camel Case, whereas "KeyVault" is Camel Case. Just a note on nomenclature.

Comment (v-fearam) [reply azure_jumpstart_arcbox/ARM/azuredeploy.json]: done

File: azure_jumpstart_arcbox/ARM/key-vault/key-vault.json (hunk 2 above)
Comment (v-fearam) [reply azure_jumpstart_arcbox/ARM/key-vault/key-vault.json]: done

File: azure_jumpstart_arcbox/bicep/main.bicep (hunk 4 above)
Comment (v-fearam) [reply azure_jumpstart_arcbox/bicep/main.bicep]: done

File: azure_jumpstart_arcbox/bicep/key-vault/key-vault.bicep (hunk 3 above)
Comment (v-fearam) [reply azure_jumpstart_arcbox/bicep/key-vault/key-vault.bicep]: done

**Review by dkirby-ms — APPROVED** [review]

**Pull request merged**